File: homework/home4.1/src/PizzaDirector.java
public class PizzaDirector {
    private final PizzaBuilder builder;

    public PizzaDirector(PizzaBuilder builder) {
        this.builder = builder;
    }

    public void addIngredient(Ingredient ingredient) {
        builder.addIngredient(ingredient);
    }

    public Pizza constructPizza() {
        return builder.buildPizza();
    }
}

File: classwork/classwork4.1/PizzaDirector.cpp
#pragma once

#include <memory>
#include "PizzaBuilder.cpp"

class PizzaDirector {
private:
    std::unique_ptr<PizzaBuilder> builder;

public:
    PizzaDirector(std::unique_ptr<PizzaBuilder> _builder) : builder(std::move(_builder)) {}

    void addIngredient(const Ingredient& ingredient) {
        builder->addIngredient(ingredient);
    }

    std::unique_ptr<Pizza> constructPizza() {
        return builder->buildPizza();
    }
};
